troposphere/static/js/actions/ImageVersionActions.js

```diff
@@ -23,8 +23,11 @@ export default {
             let err_message = error_json;
             if (error_json.hasOwnProperty("non_field_errors")) {
                 err_message = error_json.non_field_errors.join(" , ");
+                if(err_message.name  != null && err_message.name[0] != null) {
+                    err_message = err_message.name[0]
+                }
             }
-            var message = "Error updating Image Version " + version.get("name") + ": " + err_message;
+            var message = "Error updating Image Version '" + version.get("name") + "': " + err_message;
```

Comment:
I'm not referring to the specific line (or really this pr). This block of code is unmanageable. It's trying way too hard to get a value.

1) Multiple nested conditionals are hard to reason about
2) Falsy values are hard to get right
3) Bending over backwards for a weak interface is error prone

Here are the implications w.r.t to this chunk of code:
If `name` is not an array, then an error will be thrown. If non_field_errors is not a property, then [object Object] will be shown (1). 

The following test `err_message.name != null && err_message.name[0] != null` is slightly off. The real test is whether its an array with a first value. So it should be `Array.isArray(name) && name.length > 0`. The original test was whether or not the first element is falsy(2).

The most important point is 3. Looking at this code at a glance it's really hard to discern what error api it supports. It's just a bunch of edge cases. If the api is truly that non-standard, don't make the frontend also error prone. Replace the whole block with:
```
}).fail(function(err) {
            var title = "Error updating Image Version."
            var message = JSON.stringify(err.responseJSON);
            NotificationController.error(title, message);
            Utils.dispatch(ImageVersionConstants.REMOVE_IMAGE_VERSION, {
                version: version
            });
})
```

As a tradeoff I lose the few edge cases where the original code produced nice error messages. But this code is /way/ easier to read. It's also easier to verify that it won't produce errors.

In the case where we /need/ good error messages, then the fix should be applied to make a nice error api. Then both the frontend and backend will be easy to reason about and clean. A simpler alternative in troposphere is to add more validation so that you cannot fire the initial request with an empty string.